### PR TITLE
Made blue beam

### PR DESCRIPTION
**Pull request by nick-wilson95**

### CODE REVIEW

File: objects/oBlueBeam/Draw_0.gml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+var rand = random(100);
```
Comment (Drummerboy444) [inline objects/oBlueBeam/Draw_0.gml]: Should be `_num_rand`

File: objects/oBlueBeam/Draw_0.gml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+if(rand < 10) _num_width = 0;
```
Comment (Drummerboy444) [inline objects/oBlueBeam/Draw_0.gml]: `_num_width` should be declared as a local variable, at the moment it's getting created as an instance variable

File: objects/oChannelService/Step_2.gml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 if (!bool_attemptingToChannel) {
 	bool_readyToChannel = true;
 	if (bool_channelling) {
-		ChannelService_CANCEL_CHANNEL();
+		ChannelService_CANCEL_CHANNEL(num_targetX, num_targetY);
```
Comment (Drummerboy444) [inline objects/oChannelService/Step_2.gml]: You don't need to pass these variables through, as we're saving `num_targetX` and `num_targetY` on the channel service and these scripts are all private. So we can just directly access targetX and targetY from within the private channel service scripts

File: scripts/BlueBeam_CancelChannel/BlueBeam_CancelChannel.gml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#region Doc
+/// @function BlueBeam_CancelChannel(id_caster, num_targetX, num_targetY, id_ability) A template for a CancelChannel script for a channel ability.
```
Comment (Drummerboy444) [inline scripts/BlueBeam_CancelChannel/BlueBeam_CancelChannel.gml]: Left over docs from the template here

File: scripts/BlueBeam_CancelChannel/BlueBeam_CancelChannel.gml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+/// @param   {id}      id_caster		 The id of the abilityCaster.
+/// @param   {number}  num_targetX       The x target of the ability.
+/// @param   {number}  num_targetY       The y target of the ability.
+/// @param	 {id}	   id_ability		 The id of the ability object to be used by this script, generated in the StartChannel script.
```
Comment (Drummerboy444) [inline scripts/BlueBeam_CancelChannel/BlueBeam_CancelChannel.gml]: I think `id_blueBeam` might be better instead of just `id_ability`

Comment (Drummerboy444) [reply scripts/BlueBeam_CancelChannel/BlueBeam_CancelChannel.gml]: Would be good to change on the other blue beam scripts too

File: scripts/BlueBeam_CancelChannel/BlueBeam_CancelChannel.gml (hunk 4 above)
Comment (Drummerboy444) [reply scripts/BlueBeam_CancelChannel/BlueBeam_CancelChannel.gml]: Needs changing on the others too

File: scripts/BlueBeam_StartChannel/BlueBeam_StartChannel.gml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+/// @param   {number}  num_targetY       The y target of the ability.
+/// @returns {id}	   id_ability		 The id of the ability object generated in this script.
+#endregion
+
+var _id_caster = argument[0];
+var _num_targetX = argument[1];
+var _num_targetY = argument[2];
+
+// Note: we obviously won't want to return a generic abilit like this.
+var ability = instance_create_layer(_id_caster.x, _id_caster.y, "Abilities", oBlueBeam);
```
Comment (Drummerboy444) [inline scripts/BlueBeam_StartChannel/BlueBeam_StartChannel.gml]: Should be `_id_ability`

File: scripts/BlueBeam_StartChannel/BlueBeam_StartChannel.gml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+/// @param   {number}  num_targetX       The x target of the ability.
+/// @param   {number}  num_targetY       The y target of the ability.
+/// @returns {id}	   id_ability		 The id of the ability object generated in this script.
+#endregion
+
+var _id_caster = argument[0];
+var _num_targetX = argument[1];
+var _num_targetY = argument[2];
+
+// Note: we obviously won't want to return a generic abilit like this.
```
Comment (Drummerboy444) [inline scripts/BlueBeam_StartChannel/BlueBeam_StartChannel.gml]: Not sure what you mean by this comment?

File: scripts/BlueBeam_StartChannel/BlueBeam_StartChannel.gml (hunk 6 above)
Comment (Drummerboy444) [reply scripts/BlueBeam_StartChannel/BlueBeam_StartChannel.gml]: actually `_id_blueBeam` is probably better

File: scripts/ChannelService_Channel/ChannelService_Channel.gml [hunk 8]
```diff
@@ -34,21 +35,21 @@ with(_id_channelService){
 	bool_attemptingToChannel = true;
 
 	if (bool_readyToChannel) {
-		ChannelService_START_CHANNEL(_enum_ability, _num_targetX, _num_targetY);
+		ChannelService_START_CHANNEL(_enum_ability, num_targetX, num_targetY);
```
Comment (Drummerboy444) [inline scripts/ChannelService_Channel/ChannelService_Channel.gml]: Again, you don't have to pass these target arguments through, same for the other private channel scripts used in this file

File: objects/oBlueBeam/Create_0.gml [hunk 9]
```diff
@@ -0,0 +1,2 @@
+num_targetX = 0;
+num_targetY = 0;
```
Comment (Drummerboy444) [inline objects/oBlueBeam/Create_0.gml]: Maybe num_targetX and num_targetY can live on the oChannel object? It seems like every channel will want to work like this.

Comment (nick-wilson95) [reply objects/oBlueBeam/Create_0.gml]: The ability object also needs to have these set and updated somewhere that it can reference in it's draw function.

**Comment by nick-wilson95** [comment]
[Create basic channel (blue beam)](https://trello.com/c/xRTVjkY1/8-create-basic-channel-blue-beam)

**Pull request merged**